route_guide/route_guide_client.cc

```diff
@@ -23,238 +23,324 @@
-using stout::grpc::Stream;
+using stout::eventuals::Terminal;
+using stout::eventuals::Terminate;
+using stout::eventuals::grpc::Client;
+using stout::eventuals::grpc::CompletionPool;
+using stout::eventuals::grpc::Stream;
 
+// Populate Point message (you can check this message structure in
+// route_guide.proto)
 Point MakePoint(long latitude, long longitude) {
   Point p;
   p.set_latitude(latitude);
   p.set_longitude(longitude);
   return p;
 }
 
-Feature MakeFeature(const std::string& name,
-                    long latitude, long longitude) {
+// Populate Feature message
+Feature MakeFeature(const std::string& name, long latitude, long longitude) {
   Feature f;
   f.set_name(name);
   f.mutable_location()->CopyFrom(MakePoint(latitude, longitude));
   return f;
 }
 
-RouteNote MakeRouteNote(const std::string& message,
-                        long latitude, long longitude) {
+// Populate RouteNote message
+RouteNote MakeRouteNote(const std::string& message, long latitude,
+                        long longitude) {
   RouteNote n;
   n.set_message(message);
   n.mutable_location()->CopyFrom(MakePoint(latitude, longitude));
   return n;
 }
 
+// The basic idea of stout-grpc is based on the callbacks. stout-grpc is
+// intented to be a higher-level interface while still being asynchronous
+// that uses stout-eventuals (https://github.com/3rdparty/stout-eventuals)
+// to make composing async code easier.
+
 class RouteGuideClient {
  public:
+  // Populate feature_list_ vector with the data from db(route_guide_db.json)
   RouteGuideClient(const std::string& db, Client* client) : client_(client) {
     routeguide::ParseDb(db, &feature_list_);
   }
 
-  void GetFeature() {
-    Point point;
-    Feature feature;
-    point = MakePoint(409146138, -746188906);
-    GetOneFeature(point, &feature);
-    point = MakePoint(0, 0);
-    GetOneFeature(point, &feature);
-  }
-
-  void ListFeatures() {
-    Rectangle rect;
-    Feature feature;
-
-    rect.mutable_lo()->set_latitude(400000000);
-    rect.mutable_lo()->set_longitude(-750000000);
-    rect.mutable_hi()->set_latitude(420000000);
-    rect.mutable_hi()->set_longitude(-730000000);
-    std::cout << "Looking for features between 40, -75 and 42, -73"
-              << std::endl;
-
-    Notification<Status> done;
-    client_->Call<RouteGuide, Rectangle, Stream<Feature>>(
-        "ListFeatures",
-        &rect,
-        [this](auto* call, auto&& feature) {
-          if (feature) {
-            std::cout << "Found feature called "
-                      << feature->name() << " at "
-                      << feature->location().latitude()/kCoordFactor_ << ", "
-                      << feature->location().longitude()/kCoordFactor_ << std::endl;
-          } else {
-            call->Finish();
-          }
-        },
-        [&](auto*, const Status& status) {
-          done.Notify(status);
-        });
+ public:
+  // The goal of GetFeature is just to implement async unary rpc call.
```

Comment:
You can simplify sentences like this:
```suggestion
  // GetFeature demonstrates an async unary rpc call.
```